performance/single-node/builtin_atomic_tests.c

```diff
@@ -0,0 +1,323 @@
+typedef _Atomic __UINT_LEAST64_TYPE__ atomic_uint_least64_t;
+typedef _Atomic __INT_FAST8_TYPE__ atomic_int_fast8_t;
+typedef _Atomic __UINT_FAST8_TYPE__ atomic_uint_fast8_t;
+typedef _Atomic __INT_FAST16_TYPE__ atomic_int_fast16_t;
+typedef _Atomic __UINT_FAST16_TYPE__ atomic_uint_fast16_t;
+typedef _Atomic __INT_FAST32_TYPE__ atomic_int_fast32_t;
+typedef _Atomic __UINT_FAST32_TYPE__ atomic_uint_fast32_t;
+typedef _Atomic __INT_FAST64_TYPE__ atomic_int_fast64_t;
+typedef _Atomic __UINT_FAST64_TYPE__ atomic_uint_fast64_t;
+typedef _Atomic __INTPTR_TYPE__ atomic_intptr_t;
+typedef _Atomic __UINTPTR_TYPE__ atomic_uintptr_t;
+typedef _Atomic __SIZE_TYPE__ atomic_size_t;
+typedef _Atomic __PTRDIFF_TYPE__ atomic_ptrdiff_t;
+typedef _Atomic __INTMAX_TYPE__ atomic_intmax_t;
+typedef _Atomic __UINTMAX_TYPE__ atomic_uintmax_t;
+*/
+/*******************************************************************************
+ * End C11 atomic types
+ ******************************************************************************/
+
+/*******************************************************************************
+ * Begin C11 atomic memory orders
+ ******************************************************************************/
+/*
+  typedef enum
+  {
+    memory_order_relaxed = __ATOMIC_RELAXED,
+    memory_order_consume = __ATOMIC_CONSUME,
+    memory_order_acquire = __ATOMIC_ACQUIRE,
+    memory_order_release = __ATOMIC_RELEASE,
+    memory_order_acq_rel = __ATOMIC_ACQ_REL,
+    memory_order_seq_cst = __ATOMIC_SEQ_CST
+  } memory_order;
+*/
+/*******************************************************************************
+ * End C11 atomic memory orders
+ ******************************************************************************/
+
+/*******************************************************************************
+ * Begin tunables
+ ******************************************************************************/
+static atomic_int atomic_var;
+static atomic_int dummy_atomic_var;
+static int val = 0; // Needs to match above type
+static int dummy_val = 0;
+
+#define MEMORY_ORDER memory_order_acq_rel
+
+#define FIELD_WIDTH 16 // output tunable
+/*******************************************************************************
+ * End tunables
+ ******************************************************************************/
+
+/*******************************************************************************
+ * Wrappers to the 28 C11 atomic functions
+ ******************************************************************************/
+#define POST_FIX_INCREMENT dbg_printf("dummy_val = %d\n", dummy_val++);
+#define INIT \
+atomic_init(&atomic_var, 0);
+
+
+#define THREAD_FENCE \
+atomic_thread_fence(MEMORY_ORDER);
+
+
+#define SIGNAL_FENCE \
+atomic_signal_fence(MEMORY_ORDER);
+
+
+#define IS_LOCK_FREE \
+atomic_is_lock_free(&atomic_var);
+
+
+#define STORE_EXPLICIT \
+atomic_store_explicit(&atomic_var, ++val, memory_order_relaxed);
```

Comment:
It doesn't accept all memory orders. See http://en.cppreference.com/w/c/atomic/atomic_store for more details.